classes/Kohana/Log.php

```diff
@@ -225,4 +240,99 @@ public function write()
 		}
 	}
 
+	/**
+	 * Logs with an arbitrary level.
+	 *
+	 * Adds a message to the log. Replacement values must be passed in to be
+	 * replaced using Text::interpolate
+	 *
+	 *     $log->log(\Psr\Log\LogLevel::INFO, 'You look good, today.');
+	 *
+	 *     $log->log(Log::ERROR, 'Could not locate user: {username}', array(
+	 *         'username' => $username,
+	 *     ));
+	 *
+	 * @uses   Text::interpolate Inserts the replacement values into the message
+	 * @param  mixed  $level    level of message
+	 * @param  string $message  message body
+	 * @param  array  $context  values to replace in the message
+	 * @return Log
+	 */
+	public function log($level, $message, array $context = [])
+	{
+		// check if level is available
+		if (is_int($level) AND array_key_exists($level, $this->_log_levels))
```

Comment:
`$this->_log_levels[$level] === NULL` shouldn't happen, replace `array_key_exists($level, $this->_log_levels)` with `isset($this->_log_levels[$level])` for better performance